preprocessing/processing.py

```diff
@@ -0,0 +1,272 @@
+"""
+Copyright 2020 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+
+
+import os
+import sys
+from collections import Counter, defaultdict
+import json
+import numpy as np
+import pandas as pd
+
+cwd = os.getcwd()
+sys.path.append(os.path.join(cwd, "preprocessing"))
+sys.path.append(os.path.join(cwd, 'preprocessing/raw_data/top-dataset-semantic-parsing'))
```

Comment:
Why do we not need to do this for the other dataset folders?

Reply:
The `TOP` dataset comes with a script that implements a `Tree` structure for its specific annotation format. I need to import this script when processing thus the folder needs to be in `sys.path`.

Reply:
Got it, thanks!